index.js

```diff
@@ -156,6 +156,7 @@ export function execa(rawFile, rawArgs, rawOptions) {
 	setMaxListeners(Number.POSITIVE_INFINITY, controller.signal);
 
 	pipeOutputAsync(spawned, stdioStreamsGroups);
+	cleanupOnExit(spawned, options, controller);
```

Comment:
Move this earlier since it is more related to `spawned.kill()` (the next line) that the logic in `lib/stream.js`. Also, it can be done synchronously.